cmake/NanoBindMinimal.cmake

```diff
@@ -0,0 +1,26 @@
+# =============================================================================
+# minimal nanobind interface
+# =============================================================================
+
+set(NB_DIR ${PROJECT_SOURCE_DIR}/external/nanobind)
+
+set(NB_SOURCE_FILES
```

Comment:
is there a reason not to follow the `add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/ext/nanobind)` and
`nanobind_add_module`  pattern? This seems more complicated, and if/when nanobind refactors things, this list needs to be kept up to date.

IMO, the reason should be documented here, so someone doesn't try do a cleanup pass to use nanobind in the conventional way.

Reply:
There's a lot of logic to build `nrniv_lib`, it's not a straightforward python library. We basically want to use nanobind as a library but nanobind cmake is not prepared for it.

Reply:
If someone manages to use `nanobind_add_module` in the current Neuron cmake setup I would be be very pleased. I didn't find a way.

Reply:
> It's worth explicitly writing what the issues are.
It simply doesn't fit the case. `nanobind_add_module` is to create a python module. nrniv_lib is a plain library which happens to handle python calls without the python interpreter initialization (which goes in another file).

> Wouldn't it also be possible to have nanobind create another library, and just load it in the main share/lib/python/neuron/__init__.py, so that one doesn't have to worry about having it in nrniv_lib?
Possibly. That's what we kind of try to do currently with the python module entry point (`nrnpython`)
```
target_link_libraries(nrnpython${pyver} PRIVATE ${Readline_LIBRARY} ${nanobind_target})
```
But all these shared libraries are being built in a very custom fashion, it's not trivial to keep the old behavior with the provided nanobind cmake

Reply:
This looks a lot like
```
function (nanobind_build_library TARGET_NAME)
```
from the official CMake, though. Is that not reusable?

Reply:
Nope. It has several guards, both at the file and function level. Ended up getting inspiration from it.